numpy/distutils/ccompiler_opt.py

```diff
@@ -959,8 +959,12 @@ def __init__(self):
         detect_arch = (
             ("cc_on_x64",      ".*(x|x86_|amd)64.*", ""),
             ("cc_on_x86",      ".*(win32|x86|i386|i686).*", ""),
-            ("cc_on_ppc64le",  ".*(powerpc|ppc)64(el|le).*", ""),
-            ("cc_on_ppc64",    ".*(powerpc|ppc)64.*", ""),
+            ("cc_on_ppc64le",  ".*(powerpc|ppc)64(el|le).*|.*powerpc.*", ""
+                                          "defined(__LP64__) && "
```

Comment:
```suggestion
                                          "defined(__powerpc64__) && "
```
rather than testing the pointer size

Reply:
Done.